### PR TITLE
MM-36002: Remove enterprise terminology

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
This removes any mention to E10/E20. I was going to change the terminology in the code itself, too, but in order to minimize risk for v1.11.1, I only changed user-facing strings. We can change the names in the code for v1.12, no problem.

Set the base branch to https://github.com/mattermost/mattermost-plugin-incident-collaboration/pull/572 because that one also included changes in the terminology. I'll change to master as soon as https://github.com/mattermost/mattermost-plugin-incident-collaboration/pull/572 is merged.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-36002

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]

File: webapp/src/components/backstage/upgrade_modal_data.tsx [hunk 1]
```diff
@@ -116,7 +116,7 @@ export const getUpgradeModalCopy = (
                 titleText: 'Your 30-day trial has started',
                 helpText: (
                     <span>
-                        {`Your Enterprise E10 license expires on ${expiryDate}. You can purchase a license at any time through the `}
+                        {`Your Mattermost Enterprise license expires on ${expiryDate}. You can purchase a license at any time through the `}
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/upgrade_modal_data.tsx:119]: This is not a typo. The previous string was wrong, as the trial license is effectively an Enterprise (old E20) license.

File: server/api/bot.go [hunk 2]
```diff
@@ -120,8 +120,8 @@ outer:
 	post.Message = "Thank you!"
 	attachments := []*model.SlackAttachment{
 		{
-			Title: "You’re currently on a free trial of our Enterprise Edition E20 license.",
-			Text:  "Your free trial will expire in **30 days**. Visit our Customer Portal to purchase a license to continue using Enterprise Edition E10 and E20 features after your trial ends.\n[Purchase a license](https://customers.mattermost.com/signup)\n[Contact sales](https://mattermost.com/contact-us/)",
+			Title: "You’re currently on a free trial of our Enterprise Edition license.",
```
Comment (justinegeffen) [inline server/api/bot.go:123]:
```suggestion
			Title: "You’re currently on a free trial of Mattermost Enterprise.",
```

File: server/api/bot.go [hunk 3]
```diff
@@ -120,8 +120,8 @@ outer:
...
+			Text:  "Your free trial will expire in **30 days**. Visit our Customer Portal to purchase a license to continue using Professional and Enterprise features after your trial ends.\n[Purchase a license](https://customers.mattermost.com/signup)\n[Contact sales](https://mattermost.com/contact-us/)",
```
Comment (justinegeffen) [inline server/api/bot.go:124]:
```suggestion
			Text:  "Your free trial will expire in **30 days**. Visit our Customer Portal to purchase a license to continue using commercial edition features after your trial ends.\n[Purchase a license](https://customers.mattermost.com/signup)\n[Contact sales](https://mattermost.com/contact-us/)",
```

File: server/bot/poster.go [hunk 4]
```diff
@@ -107,27 +107,27 @@ func (b *Bot) NotifyAdmins(messageType, authorUserID string, isTeamEdition bool)
 
 	footer := "[Learn more](https://mattermost.com/pricing-self-managed/).\n\nWhen you select **Start 30-day trial**, you agree to the [Mattermost Software Evaluation Agreement](https://mattermost.com/software-evaluation-agreement/), [Privacy Policy](https://mattermost.com/privacy-policy/), and receiving product emails."
 	if isTeamEdition {
-		footer = "[Learn more](https://mattermost.com/pricing-self-managed/).\n\n[Convert to Enterprise Edition](https://docs.mattermost.com/install/ee-install.html#converting-team-edition-to-enterprise-edition) to unlock this feature. Then, start a trial or upgrade to Enterprise Edition E10 or E20."
+		footer = "[Learn more](https://mattermost.com/pricing-self-managed/).\n\n[Convert to Starter Edition](https://docs.mattermost.com/install/ee-install.html#converting-team-edition-to-enterprise-edition) to unlock this feature. Then, start a trial or upgrade to Mattermost Professional or Enterprise."
```
Comment (justinegeffen) [inline server/bot/poster.go:110]:
```suggestion
		footer = "[Learn more](https://mattermost.com/pricing-self-managed/).\n\n[Convert to Mattermost Starter](https://docs.mattermost.com/install/ee-install.html#converting-team-edition-to-enterprise-edition) to unlock this feature. Then, start a trial or upgrade to Mattermost Professional or Enterprise."
```

**Review by justinegeffen — COMMENTED** [review]
I've added some feedback based on the conversations we've had today.

File: server/api/bot.go (hunk 3 above)
Comment (agarciamontoro) [reply server/api/bot.go:124]: @justinegeffen, should this be "commercial edition**s**"?

Comment (justinegeffen) [reply server/api/bot.go:124]: We're referring to the features of both commercial editions so a collective "commercial edition features" seems to read more easily than highlighting them both. However, if we want to specifically refer to which features of which edition they're using, then we should probably mention them by specific name. 

Does that make sense?

Comment (agarciamontoro) [reply server/api/bot.go:124]: Oh, I see, thank you for the explanation! Adding your suggestion then :)

**Comment by justinegeffen** [comment]
Thanks for this, @agarciamontoro! :)

**Comment by agarciamontoro** [comment]
@prapti, skipping QA here as this is literally just user-facing strings changes.

**Pull request merged**